cocos/ui/UIButton.cpp

```diff
@@ -788,12 +789,17 @@ void Button::setTitleFontSize(float size)
     {
         _titleRenderer->setSystemFontSize(size);
     }
-    else
+    else if (_type == FontType::TTF)
     {
         TTFConfig config = _titleRenderer->getTTFConfig();
         config.fontSize = size;
         _titleRenderer->setTTFConfig(config);
     }
+    else
+    {
+        // BMFont can't set its font size
+        size = _fontSize;
```

Comment:
@felixjingga  We shouldn't call `updateContentSize` if the content size is not changed. But we do need to record `_fontSize`, so if we change the Font type to TTF or system font, the new font size would take effect.